contracts/example/src/event.rs

```diff
@@ -1,17 +1,40 @@
-use soroban_sdk::{Bytes, Env, String, Symbol};
+use soroban_sdk::{Address, Bytes, BytesN, String};
+use stellar_axelar_std::events::Event;
+use stellar_axelar_std::IntoEvent;
 
-pub fn executed(
-    env: &Env,
-    source_chain: String,
-    message_id: String,
-    source_address: String,
-    payload: Bytes,
-) {
-    let topics = (
-        Symbol::new(env, "executed"),
-        source_chain,
-        message_id,
-        source_address,
-    );
-    env.events().publish(topics, (payload,));
+#[derive(Debug, PartialEq, Eq, IntoEvent)]
+pub struct ExecutedEvent {
+    pub source_chain: String,
+    pub message_id: String,
+    pub source_address: String,
+    #[data]
+    pub payload: Bytes,
+}
+
+#[derive(Debug, PartialEq, Eq, IntoEvent)]
+pub struct TokenReceivedEvent {
+    pub source_chain: String,
+    pub message_id: String,
+    pub source_address: Bytes,
+    #[data]
+    pub payload: Bytes,
+    #[data]
+    pub token_id: BytesN<32>,
+    #[data]
+    pub token_address: Address,
+    #[data]
+    pub amount: i128,
```

Comment:
Stellar has no limit on number or size of topics, so these 3 can go inside topics